coding-standards-and-naming.rst

```diff
@@ -188,6 +226,192 @@ happen with the agreement of the parties involved). This document does not try
+   functions that perform some operation that may succeed or fail.
+
+Variable Names
+==============
+
+Variable names MUST be meaningful. One letter variable names SHOULD be avoided,
+except for places where the variable has no real meaning or a trivial meaning
+(e.g. ``for (i=0; i<100; i++) ...``).
+
+Variable names MUST be in lowercase. Use underscores to separate between words.
+
+Internal Function Naming Conventions
+====================================
+
+The main module source file MUST be named ``modulename.c``.
+
+Header files that are used by other sources must be named ``php_modulename.h``.
+
+Functions that are part of the external API MUST be named
+``php_modulename_function()`` to avoid symbol collision. They MUST be in
+lowercase, with words underscore delimited. Exposed API MUST be defined in
+``php_modulename.h``:
+
+.. code::
+
+   PHPAPI char *php_session_create_id(PS_CREATE_SID_ARGS);
+
+Unexposed module function MUST be ``static`` and MUST NOT be defined in
+`php_modulename.h`:
+
+.. code::
+
+   static int php_session_destroy()
+
+Syntax and indentation
+======================
+
+You SHOULD Use K&R-style. When you write code that goes into the core of PHP or
+one of its standard modules, please maintain the K&R style.
+
+Be generous with whitespace and braces. Keep one empty line between the variable
+declaration section and the statements in a block, as well as between logical
+statement groups in a block. Maintain at least one empty line between two
+functions, preferably two. Always prefer:
+
+.. code::
+
+   if (foo) {
+       bar;
+   }
+
+to:
+
+.. code::
+
+   if(foo)bar;
+
+You MUST use the tab character when indenting. A tab is expected to represent
+four spaces. It is important to maintain consistency in indentation so that
+definitions, comments, and control structures line up correctly.
+
+Preprocessor statements (``#if`` and such) MUST start at column one. To indent
+preprocessor directives you should put the ``#`` at the beginning of a line,
+followed by any number of spaces.
+
+The length of constant string literals SHOULD be calculated via ``strlen()``
+instead of using ``sizeof()-1`` as it is clearer and any modern compiler will
+optimize it away. Legacy usages of the latter style exists within the codebase
+but SHOULD NOT be refactored, unless larger refactoring around that code is
+taking place.
+
+Testing
+=======
+
+Extensions SHOULD be well tested using ``*.phpt`` tests. Read more at
+`qa.php.net documentation <https://qa.php.net/write-test.php>`_.
+
+Experimental Functions
+======================
+
+To reduce the problems normally associated with the first public implementation
+of a new set of functions, it has been suggested that the first implementation
+SHOULD place a file labeled `EXPERIMENTAL` in the function directory. Functions
+MUST follow the standard prefixing conventions during their initial
+implementation.
+
+The file labelled ``EXPERIMENTAL`` SHOULD include the following information:
```

Comment:
1. It's unclear to me whether this refers to internal or user functions.
2. We have never done this in practice, and I think we should completely drop this section.

Reply:
We had a lot of extensions marked as `EXPERIMENTAL` in the olden days, but that should no longer happen. Only stable extensions should be added to php-src. New functions go into master only, and master is somewhat "experimental" anyway, so I agree that there is no need for this section.

Reply:
I agree. I think we should rewrite this section, and rewrite it to:

```
New extensions MUST start out as third-party extensions, or in an experimental branch, until an RFC is passed to add them to the core distribution.
```